.github/workflows/do-not-merge-label.yml

```
@@ -0,0 +1,23 @@
+name: Check do-not-merge labels
+on:
+  pull_request:
+    types:
+      - opened
+      - reopened
+      - labeled
+      - unlabeled
+      - synchronize
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    name: Check do-not-merge labels
+    steps:
+      - name: Get PR labels
+        id: pr-labels
+        uses: joerick/pr-labels-action@v1.0.9
+
+      - run: |
+          echo "A 'do-not-merge/*' label has been found in this PR."
+          exit 1
+        if: contains(steps.pr-labels.outputs.labels, 'do-not-merge')
```

Possible nit (doesn't really matter in reality so feel free to ignore)
```suggestion
        if: startsWith(steps.pr-labels.outputs.labels, 'do-not-merge')
```

---

It probably doesn't work I guess, because it would need to be applied to all labels, which this does not. Again, don't take this suggestion too seriously.

---

That's the output of the workflow: 

```
Found label  changelog/no-changelog 
  Setting env var for remaining steps: GITHUB_PR_LABEL_CHANGELOG_NO_CHANGELOG=1

Found label  dev/testing 
  Setting env var for remaining steps: GITHUB_PR_LABEL_DEV_TESTING=1

Found label  dev/tooling 
  Setting env var for remaining steps: GITHUB_PR_LABEL_DEV_TOOLING=1

Action output:
labels: " changelog-no-changelog dev-testing dev-tooling "
labels-object: {"changelog-no-changelog":true,"dev-testing":true,"dev-tooling":true}
```

I use labels to avoid parsing a json, not sure that would really be needed here. So I'll keep contains